File: argentum-api/src/main/java/net/devium/argentum/rest/model/request/CancelOrderItemRequest.java
package net.devium.argentum.rest.model.request;

import java.math.BigDecimal;

public class CancelOrderItemRequest {
    // Union order ID if customCancelled is used and cancelled is unused.
    private Long id;
    private int cancelled;
    private BigDecimal customCancelled;

    public Long getId() {
        return id;
    }

    public int getCancelled() {
        return cancelled;
    }

    public BigDecimal getCustomCancelled() {
        return customCancelled;
    }
}

File: argentum-ui/src/app/common/rest-service/request/cancel-order-item-request.ts
import { OrderItem } from '../../model/order-item';
import { Order } from '../../model/order';

export class CancelOrderItemRequest {
  id: number;
  cancelled: number;
  customCancelled: number;
}

export function cancelFromOrderItem(orderItem: OrderItem): CancelOrderItemRequest {
  return {
    id: orderItem.id,
    cancelled: orderItem.cancelled,
    customCancelled: null
  };
}

export function cancelFromOrder(order: Order): CancelOrderItemRequest {
  return {
    id: order.id,
    cancelled: null,
    customCancelled: order.customCancelled
  };
}
